src/core/store/middleware.ts

```diff
@@ -96,11 +97,12 @@ export namespace Middleware {
     }
 
     return compose(
-      thunkEvaluator,
-      saveStateAnalyzer,
+      applyMiddleware(thunkEvaluator, saveStateAnalyzer),
       reduxBatch,
       applyMiddleware(...middleware),
-      reduxBatch
+      reduxBatch,
+      applyMiddleware(thunkEvaluator),
+      reduxBatch,
```

Comment:
Why do we need to call reduxBatch, then thunk, then reduxBatch? If the action is a batchAction, then it'll go through the whole middleware stack then, wouldn't it? Would it not suffice to call thunkEvaluator then reduxBatch?